aom/src/main/java/com/nedap/archie/aom/utils/AOMUtils.java

```diff
@@ -197,39 +197,42 @@ public static boolean codeExistsAtLevel(String nodeId, int specializationDepth)
         return false;
     }
 
-    public static boolean archetypeIdMatchesSlotExpression(String archetypeRef, ArchetypeSlot slot) {
-        boolean includePasses = false;
-
-        if(slot.getIncludes() != null && !slot.getIncludes().isEmpty()) {
+    public static boolean archetypeRefMatchesSlotExpression(String archetypeRef, ArchetypeSlot slot) {
+        if(!isSlotPresentAndOpen(slot.getIncludes()) && isSlotPresentAndOpen(slot.getExcludes())) {
             for (Assertion include : slot.getIncludes()) {
-                if (filterInclude(include.getExpression(), archetypeRef)) {
-                    includePasses = true;
+                if (matchesInclude(include.getExpression(), archetypeRef)) {
+                    return true;
                 }
             }
-        } else {
-            includePasses = true;
-        }
-        if(includePasses) {
-            if (slot.getExcludes() != null && !slot.getExcludes().isEmpty()) {
-                for (Assertion exclude : slot.getExcludes()) {
-                    if (filterExclude(exclude.getExpression(), archetypeRef)) {
-                        return false;
-                    }
+            return false;
+        } else if (!isSlotPresentAndOpen(slot.getExcludes()) && isSlotPresentAndOpen(slot.getIncludes())) {
+            for (Assertion exclude : slot.getExcludes()) {
+                if (matchesExclude(exclude.getExpression(), archetypeRef)) {
+                    return false;
                 }
             }
         }
-        return includePasses;
+        return true;
```

Comment:
If both the include and exclude at this point are present but not open (which is not allowed), is that at this point already validated or does this method just assume that will never be the case?

Reply:
This will result in the archetype not validating. See [this example](https://github.com/openehr/archie/blob/master/tools/src/test/resources/adl2-tests/validity/slots/openEHR-EHR-SECTION.VDSEV_slot_include_not_any_exclude_not_any.v1.0.0.adls) for a test case that runs as part of the BigArchetypeValidator, and see VariousStructureValidation.java for the implementation of the `VDSEV`-validation.

So, this situation could still happen, but does not mean anything. In that case, there's the choice of here returning true or false, or throwing an exception. The exception seemed like a not so good idea - you don't want this method to fail and mess up whatever you were doing. Returning false also seems like a less good idea - it prevents people from working with these archetypes. So returning true was the choice I made.

Reply:
Ok!